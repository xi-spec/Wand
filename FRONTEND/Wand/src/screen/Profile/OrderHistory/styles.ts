import { StyleSheet } from 'react-native';
import { FONTS, SIZE } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center'
  },
  shadowBox: {
    width: '90%',
    height: '15%',
    justifyContent: 'center',
    shadowColor: 'grey',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.15,
    shadowRadius: 3.84,

    elevation: 5,
    backgroundColor: 'white',
    marginVertical: 10,
    borderRadius: 5
  },
  rowContainer: {
    flexDirection: 'row'
  },
  title: {
    fontFamily: FONTS.primaryMedium,
    fontSize: 16,
    marginHorizontal: 20
  },
  text: {
    fontFamily: FONTS.primary,
    fontSize: SIZE.second
  }

});

export default styles;
