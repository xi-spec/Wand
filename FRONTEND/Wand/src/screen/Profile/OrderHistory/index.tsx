import React from 'react';
import { Text, View } from 'react-native';
import { Order } from '../../../models/initiaState';
import styles from './styles';
import { connect } from 'react-redux';

function OrderHistory ({ orderHistory, navigation }:{orderHistory:Order[], navigation:{navigate:Function}}) {
  function totalIem (params) {
    return params.reduce((acc, product) => acc + product.quantity, 0);
  }

  return (
      <View style={styles.container}>
       {orderHistory && orderHistory.map(order => (
         <View style={styles.shadowBox}key={order._id}>
         <View style={styles.rowContainer}>
         <Text style={styles.title}>Order id:</Text>
         <Text style={styles.text}>{order._id}</Text>
         </View>
         <View style={styles.rowContainer}>
         <Text style={styles.title}>Date:</Text>
         <Text style={{ ...styles.text, marginLeft: 30 }}>{order.creationDate.replace('T', '-')}</Text>
         </View>
         <View style={styles.rowContainer}>
         <Text style={styles.title}>Total item:</Text>
         <Text style={styles.text}>{totalIem(order.cartList)}</Text>
         </View>
         <View style={styles.rowContainer}>
         <Text style={styles.title}>Total cost:</Text>
         <Text style={styles.text}>{`${order.totalCost.toFixed(2)} euros`}</Text>
         </View>
         </View>
       ))}
      </View>
  );
}

function mapStateToProps ({ user: { orderHistory } }) {
  return {
    orderHistory
  };
}
export default connect(mapStateToProps)(OrderHistory);
