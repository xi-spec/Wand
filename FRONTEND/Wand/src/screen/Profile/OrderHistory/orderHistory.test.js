import React from 'react';
import { render } from '@testing-library/react-native';
import OrderHistory from './index';
import ConfigureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('Given a OrderHistory component', () => {
  let user;
  let navigation;
  const mockStore = ConfigureStore();
  const store = mockStore({
    user: {
      orderHistory: [
        { _id: 1, cartList: [{ quantity: 1 }], totalCost: 1, creationDate: '222222TTTTT' }
      ]
    }
  });

  beforeEach(() => {
    navigation = {
      goBack: jest.fn(),
      navigate: jest.fn()
    };
  });

  describe('When is render', () => {
    test('Then is rendered correctly', () => {
      const tree = render(<Provider store={store}><OrderHistory navigation={navigation} user={user}/></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
});
