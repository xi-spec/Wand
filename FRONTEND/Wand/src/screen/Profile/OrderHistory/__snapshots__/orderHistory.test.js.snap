// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a OrderHistory component When is render Then is rendered correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 5,
        "elevation": 5,
        "height": "15%",
        "justifyContent": "center",
        "marginVertical": 10,
        "shadowColor": "grey",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3.84,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 16,
            "marginHorizontal": 20,
          }
        }
      >
        Order id:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 14,
          }
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 16,
            "marginHorizontal": 20,
          }
        }
      >
        Date:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 14,
            "marginLeft": 30,
          }
        }
      >
        222222-TTTT
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 16,
            "marginHorizontal": 20,
          }
        }
      >
        Total item:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 14,
          }
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 16,
            "marginHorizontal": 20,
          }
        }
      >
        Total cost:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 14,
          }
        }
      >
        1.00 euros
      </Text>
    </View>
  </View>
</View>
`;
